tpi_rs/src/utils/mod.rs

```diff
@@ -20,3 +20,46 @@ pub fn logging_sink<T: Display + Send + 'static>(
         }
     })
 }
+
+pub fn string_from_utf16(bytes: &[u8], little_endian: bool) -> String {
+    let u16s = bytes.chunks_exact(2).map(|pair| {
+        let Ok(owned) = pair.try_into() else {
+            unreachable!()
+        };
+
+        if little_endian {
+            u16::from_le_bytes(owned)
+        } else {
+            u16::from_be_bytes(owned)
+        }
+    });
+
+    let mut string = char::decode_utf16(u16s)
```

Comment:
[from_utf16_lossy](https://doc.rust-lang.org/alloc/string/struct.String.html#method.from_utf16_lossy) does the same